sale_line_price_properties_based/product.py

```
@@ -0,0 +1,75 @@
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductProduct(orm.Model):
+    _inherit = "product.product"
+
+    _columns = {
+        'price_formula_id': fields.many2one(
+            'mrp.property.formula', 'Price formula',
+            help="You can use the variables\n"
+                 " - self\n"
+                 " - cr\n"
+                 " - uid\n"
+                 " - ptype\n"
+                 " - product_id\n"
+                 " - properties (dictionary of properties)\n"
+                 "You have to put the result in the 'result' variable"),
+    }
+
+
+class ProductTemplate(orm.Model):
+    _inherit = "product.template"
+
+    def _price_get(self, cr, uid, products, ptype='list_price', context=None):
+        if context is None:
+            context = {}
+
+        res = super(ProductTemplate, self)._price_get(
+            cr, uid, products, ptype=ptype, context=context)
+        if 'properties' in context:
+            for product in products:
+                res = super(ProductTemplate, self)._price_get(
+                    cr, uid, products, ptype=ptype, context=context)
+                if product.price_formula_id:
+                    localdict = {
+                        'self': self,
+                        'cr': cr,
+                        'uid': uid,
+                        'ptype': ptype,
+                        'product_id': product.id,
+                        'properties': context['properties'],
+                    }
+                    try:
+                        res[product.id] = product.price_formula_id.\
+                            compute_formula(localdict)
+                    except ValueError as e:
+                        _logger.warning(
+                            "Formula evaluation error: '%s'" % e.message)
```

I'm not sure it's a good idea to silence the error here. Since the user  can edit the formula, he should get some feedback about evaluation errors without having to browse the server logs.